base_layer/core/src/proof_of_work/monero_rx.rs

```diff
@@ -103,16 +176,99 @@ fn monero_difficulty_calculation(header: &BlockHeader) -> Result<Difficulty, Mer
     }
 }
 
-fn create_input_blob(_data: &MoneroData) -> Result<String, MergeMineError> {
-    // Todo deserialize monero data to create string for  randomX vm
-    // Returning an error here so that difficulty can return 0 as this is not yet implemented.
-    Err(MergeMineError::HashingError)
+/// Appends merge mining hash to a Monero block and returns the Monero blocktemplate_blob
+pub fn append_merge_mining_tag(block: &MoneroBlock, hash: Hash) -> Result<String, MergeMineError> {
+    #[cfg(feature = "monero_merge_mining")]
+    {
+        let mut monero_block = block.clone();
+        let mm_tag = SubField::MergeMining(VarInt(0), hash);
+        monero_block.miner_tx.prefix.extra.0.push(mm_tag);
+        let serialized = serialize::<MoneroBlock>(&block);
+        Ok(hex::encode(&serialized).into())
+    }
+    #[cfg(not(feature = "monero_merge_mining"))]
```

Comment:
Can we not exclude this method/mod from the compilation entirely if `monero_merge_mining` is not enabled?

Reply:
Should we not rather move towards the removal of the `monero_merge_mining` feature. For the next test net, this should be standard as part of core.